doc/developer_guide.md

```diff
@@ -0,0 +1,213 @@
+# How to contribute
+
+We welcome contribuors.
+
+Just keep in ming that this project is maintained by people working in their free time. We are usually
+quite reactive, but it may happen that a contribution stays without feedback for a while.
+Don't be surprised and if you don't get feedback withing one month, don't hesitate to
+signal it in the issue.
+
+
+## Report issues
+
+You are welcome to report any issues through the GitHub interface. It is even
+better if you can provide a Pull Request fixing the problem.
+
+
+## Pull Requests
+
+Pull Requests are welcome. To be merged, the Pull Request shall be validated by:
+
+1. Passing the Continuous Integration successfully (see next section).
+2. Being reviewed by a PyQt5-stubs maintainer
+
+Some suggestions to make a good PR:
+* describe clearly the problem you are solving
+* if you can add a test, it is even better (see next section)
+* please update the CHANGELOG.md with a new entry describing your PR
+
+
+## Continuous Integration
+   
+The CI runs automatically on every pull requests and every night on the main branch. It 
+runs on all supported Python version on Linux, Windows and MacOs.
+
+The CI is based on `tox`, you can check the `tox.ini` file to see what is actually executed.
+
+The big steps are:
+* run our tests with `pytest`. This runs the tests as python scripts and verifies them with mypy.
+* run `mypy` against the PyQt5 and PyQt5-stubs.
+* check the PyQt5-stubs content against the actual PyQt5 modules. We rely on the tool `stubtest.py` (from
+  the mypy project). The checks are quite extensive and should spot any missing or non-existing attributes.
+  
+Some errors reported by `stubtest.py` can not be fixed because they reflect the design of Qt5 or
+are a limitation of static typing capabilities. In these case, it is OK to silence them,
+see the file `limitations.md` for details.
+
+
+## Running the tests locally
+
+When you improve the stubs, it is convenient to run the tests manually before submitting
+a Pull Request and running the CI.
+
+### Setup for tests
+
+You need the following:
+* a virtual environment with your local PyQt5-stubs installed
+* all PyQt5 packages installed
+* pytest and mypy
+
+First, create a dedicated virtual envonment for running the tests and activate it.
+
+    .../PyQt5-stubs/ $ python -m venv env_for_tests
+    .../PyQt5-stubs/ $ env_for_tests/Scripts/activate
```

Comment:
I would expect that it would be easier to maintain code blocks, at least the large ones, using triple backticks rather than indendation.  View the source of this comment to see the two options.

    indented line 1
    indented line 2

```
triple backticked line 1
triple backticked line 2
```